lib/internal/modules/cjs/loader.js

```diff
@@ -297,8 +327,58 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
+// This only applies to requests of a specific form:
+// 1. name/.*
+// 2. @scope/name/.*
+const EXPORTS_PATTERN = /^((?:@[^./@\\][^/@\\]*\/)?[^@./\\][^/\\]*)(\/.*)$/;
+function resolveExports(nmPath, request, absoluteRequest) {
+  // The implementation's behavior is meant to mirror resolution in ESM.
+  if (experimentalExports && !absoluteRequest) {
+    const [, name, expansion] =
+      request.match(EXPORTS_PATTERN) || [];
+    if (!name) {
+      return path.resolve(nmPath, request);
+    }
+
+    const basePath = path.resolve(nmPath, name);
+    const pkgExports = readExports(basePath);
+
+    if (pkgExports != null) {
+      const mappingKey = `.${expansion}`;
+      const mapping = pkgExports[mappingKey];
+      if (typeof mapping === 'string') {
+        return path.resolve(basePath, mapping);
+      }
```

Comment:
can we throw if it isn't a string so that we might be able to use non-string values in the future?

Reply:
I think I want to do that when I look into unifying this logic between CJS and ESM. Right now it's two independent copies. One thing that's definitely missing here is supporting arrays of strings. This also feeds into key and value validation in general (what to reject and ignore vs. throw) which is still an unsolved TODO in both implementations.

Reply:
ok to put off for now then, but to keep in mind